index.html

```diff
@@ -4001,76 +4017,84 @@ <h4>Verify Value Category</h4>
+										href="https://infra.spec.whatwg.org/#ordered-set">ordered set</a>.</p>
 							</li>
-							<li>
+
+							<li id="reslist-readingorder">
 								<p><a href="https://infra.spec.whatwg.org/#set-iterate">For each</a>
-									<var>resource</var> of <var>readingOrder</var>, set <var>uniqueResources</var> to the
-									result of running <a>append URL</a> given <var>resource</var> and
-									<var>uniqueResources</var>.</p>
+									<var>resource</var> of <var>readingOrder</var>, set <var>uniqueResources</var> to
+									the result of running <a>append URL</a> given <var>resource</var> and
+										<var>uniqueResources</var>.</p>
 							</li>
-							<li>
+
+							<li id="reslist-resources">
 								<p><a href="https://infra.spec.whatwg.org/#set-iterate">For each</a>
 									<var>resource</var> of <var>resources</var>, set <var>uniqueResources</var> to the
 									result of running <a>append URL</a> given <var>resource</var> and
-									<var>uniqueResources</var>.</p>
+										<var>uniqueResources</var>.</p>
 							</li>
-							<li>
+
+							<li id="reslist-end">
 								<p>Return <var>uniqueResources</var>.</p>
 							</li>
 						</ol>
-						
+
 						<section id="append-url">
 							<h5>Append URL</h5>
-							
+
 							<p>To <dfn>append URL</dfn> of <a><code>LinkedResource</code></a>
-								<var>resource</var> to the set <var>uniqueResources</var>, run the following steps:</p>
-							
+								<var>resource</var> to the set <var>uniqueResources</var>, for the term <var>term</var>,
+								run the following steps:</p>
+
 							<ol>
-								<li>
+								<li id="append-parse-url">
 									<p>let <var>url</var> be the result of running the <a
-										href="https://url.spec.whatwg.org/#concept-url-serializer">URL
-										serializer</a>&#160;[[!url]] on <var>resource["url"]</var> with the <var>exclude
-											fragment flag</var> set.</p>
+											href="https://url.spec.whatwg.org/#concept-url-serializer">URL
+											serializer</a>&#160;[[!url]] on <var>resource["url"]</var> with the
+											<var>exclude fragment flag</var> set.</p>
 								</li>
-								<li>
-									<p><a href="https://infra.spec.whatwg.org/#set-append">append</a>
+								<li id="append-no-dupes">
+									<p>if <var>uniqueResources</var>
+										<a href="https://infra.spec.whatwg.org/#list-contain">contains</a>
+										<var>url</var>, <a>validation error</a>. Otherwise, <a
+											href="https://infra.spec.whatwg.org/#set-append">append</a>
 										<var>url</var> to <var>uniqueResources</var>.</p>
 								</li>
-								<li>
+								<li id="append-alternates">
 									<p>if <var>resource["alternate"]</var> is set, set <var>uniqueResources</var> to the
 										result of <a>append URL</a> given <var>resource["alternate"]</var> and
-										<var>uniqueResources</var>.</p>
+											<var>uniqueResources</var>.</p>
 								</li>
```

Comment:
Not exactly... <var>resource["alternate"]</var> is an array of linked resources, ie, it is not a valid argument type to invoke `append URL`. Instead, the function must be invoked, recursively, on each element of <var>resource["alternate"]</var>.

(I should have checked this yesterday :-(